### PR TITLE
Refactor/evm editor

### PR DESCRIPTION
**Pull request by akira02**
## Summary
Whole EVM editor was rewritten to fit our sdk testing purpose.

## Related Links

**Asana**:  

**Mockup**:  

## Checklist

- [ ] Pasted Asana or Mockup link.
- [ ] Tagged labels.

### CODE REVIEW

File: src/scripts/evm/Contract.ts [hunk 1]
```diff
@@ -87,127 +48,91 @@ export const getValue2 = {
...
     },
   }),
-  args: [
-    {
-      type: ArgTypes.Number,
-      comment: "Input below 100 can trigger an error",
-      name: "a",
-    },
-  ],
+  args: [{ placeholder: "value", value: 0 }],
```
Comment (moojing) [inline src/scripts/evm/Contract.ts:137]: can we add a placeholder for quick understanding like it was ?

```
value (Input below 100 to trigger an error)
```

File: src/utils/isValidSignature.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  requestObject: EthereumTypes.EIP1193RequestPayload | undefined,
+  signature: string,
+  address?: string | null
+): Promise<boolean> => {
+  if (!requestObject || !signature || !address) throw new Error("Invalid args");
+  const { method, params } = requestObject;
+  try {
+    if (method === "personal_sign") {
+      const challenge = params?.[0];
+      console.log("personal_sign", challenge, signature, address, dappAuth);
```
Comment (moojing) [inline src/utils/isValidSignature.ts:22]: do we need this console.log

File: src/utils/isValidSignature.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    const isAuthorizedSigner = magicValue === ERC1271_MAGIC_VALUE;
+
+    return isAuthorizedSigner;
+  } catch (e) {
+    console.error(e);
+    return false;
+  }
+};
+
```
Comment (moojing) [inline src/utils/isValidSignature.ts:68]: it's great to check the correctness of the signature!

File: src/components/EvmEditors/ParamEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          size="xs"
+          colorScheme="blue"
+          onClick={() => {
+            setParam((prev) => [...prev, ["", ""]]);
+          }}
+        />
+      </Flex>
+
+      <Flex flexDir="column" mt={2}>
+        {param?.map((p, i) => (
```
Comment (moojing) [inline src/components/EvmEditors/ParamEditor.tsx:32]: could you replace the `param` with `params` and replace the p with `param`? 
or I think it's a bit hard to read these lines of code.

Comment (moojing) [reply src/components/EvmEditors/ParamEditor.tsx:32]: (`e` for `event` is okay cuz I think it's more like a convention.)

File: src/components/EvmEditors/EvmContractEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+              setMethodArgs((prev) => [...prev, ""]);
+            }}
+          />
+        </Flex>
+        <Flex flexDir="column" mt={2}>
+          {methodArgs?.map((p, i) => (
+            <Flex key={i} my="5px" alignItems="center">
+              <Textarea
+                rows={1}
+                value={p}
```
[the remark this answers is not in the document]
Comment (akira02) [reply src/components/EvmEditors/EvmContractEditor.tsx:214]: Sure that will be great

File: src/components/EvmEditors/EvmContractEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+      if (requestMethod === "eth_call") {
+        setRequestObject({
+          method: requestMethod,
+          params: [
+            {
+              from: account,
+              to: contractAddress,
+              data: data,
+            },
+            "latest",
```
[the remark this answers is not in the document]
Comment (akira02) [reply src/components/EvmEditors/EvmContractEditor.tsx:89]: It's a required parameter for eth_call method. I'll add a comment here.
https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_call

File: src/components/EvmEditors/ParamEditor.tsx (hunk 4 above)
Comment (akira02) [reply src/components/EvmEditors/ParamEditor.tsx:32]: Great idea

File: src/scripts/evm/Contract.ts (hunk 1 above)
Comment (akira02) [reply src/scripts/evm/Contract.ts:137]: Ok shoude be

File: src/utils/isValidSignature.ts (hunk 2 above)
Comment (akira02) [reply src/utils/isValidSignature.ts]: removed

**Review by moojing — APPROVED** [review]

**Pull request merged**